components/application/index.jsx

```diff
@@ -8,48 +9,50 @@ import ActivitySelect from './activity-select'
 import Head from './head'
 import Logout from './logout'
 
-const Application = ({
-  children,
-  dataProvider,
-  pathname,
-  variant = 'public', // 'public' or 'internal'
-  isAuthenticated = false,
-  ...restProps
-}) => (
-  <>
-    <Head />
-    <Container variant={variant} {...restProps}>
-      <LoadingBar fixed />
-      <AppBar>
-        {isAuthenticated ? (
-          <>
-            {dataProvider && <RepositorySelect value={dataProvider} />}
-            <Logout />
-          </>
-        ) : null}
-      </AppBar>
-      {variant === 'internal' && (
-        <SideBar tag="nav">
-          <h2 className="sr-only">Navigate your data</h2>
-          {dataProvider && (
-            <ActivitySelect>
-              {activities.routes
-                .filter((activity) => activity.parent == null)
-                .map(({ path, test }) => (
-                  <ActivitySelect.Option
-                    key={path}
-                    value={path}
-                    selected={test(pathname)}
-                    dataProviderId={dataProvider.id}
-                  />
-                ))}
-            </ActivitySelect>
-          )}
-        </SideBar>
-      )}
-      {children && <Main>{children}</Main>}
-    </Container>
-  </>
+const Application = observer(
```

Comment:
What do you actually observe? Would not `useObserver` be better?

Reply:
useObserver is for making thing observable, isn't it? I just wanted to react on the already observable item.

Reply:
No, it's a hook that accepts the function, which extracts everything you need from an observable and returns back to you.

It needs a function because of MobX can only track subscriptions from function. However, the interface is exactly the same as `useCallback`, i.e. you pass a function and get the data.

Check the example in the [documentation](https://mobx-react.js.org/observer-hook).

I asked because including a hook would produce fewer changes in git and might keep the code more readable.

Reply:
Also, if it would be useful, check the documentation, [how MobX reacts on observables](https://mobx.js.org/best/react.html#common-pitfall-consolelog).